### ISSUE TITLE
replay_buffer.py: stale next_states tensor not cleared on terminal transitions (next_state=None)

### ISSUE THREAD

**Issue by ArgonZapan** [ArgonZapan/DQN-transformer#14]
## Bug Description — Silent Data Contamination

In both `python/training/replay_buffer.py` and `python/training/prioritized_buffer.py`, when a terminal transition is added with `next_state=None`, the corresponding entry in the pre-allocated `next_states` tensor is **not cleared**:

```python
# replay_buffer.py:52-54
if next_state is not None and tf_name in next_state:
    self.next_states[key][idx] = torch.tensor(next_state[tf_name], dtype=torch.float32)
# If next_state is None → next_states[key][idx] is NOT overwritten — stale value remains
```

Since these are pre-allocated tensors using a circular buffer, the same indices are reused. If the buffer wraps around and a terminal transition happens to occupy an index previously filled with valid data, that stale data persists until the index is overwritten with a non-terminal transition.

## Why This Matters

While `done=True` in a terminal transition means the TD target ignores `next_q_values` (target = reward only), the **PER priority update still uses `td_errors`** computed from those next states. Additionally, if the buffer is ever used for purposes beyond the standard training loop (e.g., evaluation, visualization, or as a dataset for a secondary model), stale tensors would produce incorrect results silently.

## Suggested Fix

Explicitly zero out the `next_states` tensor when `next_state=None`:

```python
# replay_buffer.py:52-56
if next_state is not None and tf_name in next_state:
    self.next_states[key][idx] = torch.tensor(next_state[tf_name], dtype=torch.float32)
else:
    # Terminal transition — clear stale next state
    seq_len = self.timeframe_sizes[key]
    num_features = self.num_features
    self.next_states[key][idx] = torch.zeros(seq_len, num_features, dtype=torch.float32)
    if self.use_pin:
        self.next_states[key][idx] = self.next_states[key][idx].pin_memory()
```

Or, initialize the entire tensor to zeros at construction (currently already done), so un-overwritten entries default to zero rather than stale values.

## Labels

bug, data-quality

**Comment by ArgonZapan** [comment]
Fixed: `next_state=None` writes `torch.zeros` to the pre-allocated tensor (replay_buffer.py line 166-167).
